support-frontend/assets/helpers/abTests/__tests__/abtestTest.ts

```diff
@@ -635,8 +638,20 @@ it('targetPage matching', () => {
 		targetPageMatches('/uk/contribute/checkout', genericCheckoutOnly),
 	).toEqual(false);
 	expect(targetPageMatches('/uk/checkout', genericCheckoutOnly)).toEqual(true);
-	expect(targetPageMatches('/uk/thank-you', genericCheckoutOnly)).toEqual(true);
+	expect(
+		targetPageMatches(
+			'/uk/thank-you?product=SupporterPlus&ratePlan=Monthly&userType=current',
+			genericCheckoutOnly,
+		),
+	).toEqual(true);
```

Comment:
Good to add the test 👍 , we could also add a test for a `false` condition, eg. testing the `genericCheckoutOnly` regex with the `thank-you` page URL without the `product` query string param, `targetPageMatches` should return `false` in this scenario.

Reply:
Good idea, added `genericCheckoutOnly` & `oneTimeCheckoutOnly` regex with the `thank-you `page URL only. Thanks.